packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -18,6 +18,13 @@ export async function createPackageJson(
     ...rootPackageJson.devDependencies,
   };
 
+  const pickedDependencies = options.dependencies.reduce((acc, value) => {
+    if (allWorkspaceDeps[value]) {
+      acc[value] = allWorkspaceDeps[value];
+    }
```

Comment:
Can we log out an error if the dependency is missing?  You can use `context.logger.error`.